cmd/util/cmd/diff-states/cmd.go

```diff
@@ -398,6 +417,15 @@ func diffAccount(
 			accountRegisters1,
 			accountRegisters2,
 			common.AllStorageDomains,
+			func(address common.Address, domain common.StorageDomain, key any) bool {
+				if flagExcludeRandomBeaconHistory {
+					if isRandomBeaconHistory(serviceAccountAddress, address, domain, key) {
+						log.Info().Msgf("excluding random beacon history in account %s, domain %s, key %v", address, domain.Identifier(), key)
+						return false
+					}
+				}
+				return true
+			},
```

Comment:
Maybe move this into a top-level function with a descriptive name, like `func excludeRandomBeaconHistory(...`, and then move the check of the flag (`flagExcludeRandomBeaconHistory`) outside of the function, and depending on the flag, pass `excludeRandomBeaconHistory` or a function that constantly returns true (e.g. `func alwaysIncludeValue(...) bool { return true }`).

That way there is only one single branch at the start of the function, instead for each single value that is checked.

Reply:
Good idea!  Done in PR #7016 because this PR got auto-merged.